net/server_test.go

```diff
@@ -247,6 +250,27 @@ func TestDocQueue(t *testing.T) {
 	q.mu.Unlock()
 }
 
+func getHead(ctx context.Context, db client.DB, docID client.DocID) (cid.Cid, error) {
```

Comment:
todo: We shouldn't have a duplicate `getHead` function, with its own implementation, since there is already all this functionality within the `merkle/clock` package to be able to get a specific head CID of a document. 

Its not the nicest to use, but if there is going to be some util function that takes a store and a docID and returns the head CID, it should be along side the current implementation in the other package.

Reply:
Note: We also probably don't need the updated `Headstore` method on the `client.DB` interface, since the `datastore.Multistore` already has this method, and the entire `client.DB` interface shouldn't be a argument to this function just to get a head CID of a document.

Reply:
`net` package test code should not be referencing the `merkle/clock` package.  If I had to chose between duplicating code and referencing an internal method, I would overwhelmingly support duplicating the code.

Reply:
There are a few problems here. One is like Andy mentioned, needing to deal with `internal/merkle/clock` for `net` tests is not ideal. Also, before this PR, the `Headstore` was only available through `MultiStore` with an explicit `txn` which those test don't use. Saying the entire `client.DB` shouldn't be an argument to this function is correct and that is something I want to fix in a lot of places in our codebase. We often use large interfaces when a locally specific one would be more appropriate.

I'd rather not get into this within this PR but I will try to improve the `Headstore` utilities shortly.

Reply:
Issue created: https://github.com/sourcenetwork/defradb/issues/2621

Reply:
Usage within a test or not doesn't make much difference here. This function is reimplementing a somewhat sensitive flow and is relying on the behavior of getHead to be inline with the DB. 

If somewhat makes changes to the actual headset or head store for whatever reason, it's not necessarily going to be clear when they run the tests and the net package tests are failing. 

The fact that the merkleclock is internal or not also has no bearing.

We have a specific implementation, that is rather core to the entire correctness of the DB with the headset implementation, and that behavior must be consistent.

Reply:
We agree that it needs to be consistant. This will be fixed in a subsequent PR.

Reply:
Yupp perfect

(Note my comment was primarily in response to that it's a test and it's ok to duplicate the functionality, not the need for it to be in the pr)

Cheers